<!DOCTYPE html><html lang="en" ><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="generator" content="Jekyll v4.2.2" /><meta property="og:title" content="Statistical Learning for Multi-omics Data Analysis with R" /><meta property="og:locale" content="en" /><meta name="description" content="This post is written after taking lecture “Multi-omics Data Analysis” of 16th Asian Institute in Statistical Genetics and Genomics Workshop." /><meta property="og:description" content="This post is written after taking lecture “Multi-omics Data Analysis” of 16th Asian Institute in Statistical Genetics and Genomics Workshop." /><link rel="canonical" href="https://dlekdud2502.github.io/dlekdud2502/posts/Statistical-Learning-for-Multi&-45;omics-Data-Analysis-with-R/" /><meta property="og:url" content="https://dlekdud2502.github.io/dlekdud2502/posts/Statistical-Learning-for-Multi&-45;omics-Data-Analysis-with-R/" /><meta property="og:site_name" content="Dayoung Lee" /><meta property="og:type" content="article" /><meta property="article:published_time" content="2022-07-26T14:00:00+09:00" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Statistical Learning for Multi-omics Data Analysis with R" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"BlogPosting","dateModified":"2022-07-26T15:34:00+09:00","datePublished":"2022-07-26T14:00:00+09:00","description":"This post is written after taking lecture “Multi-omics Data Analysis” of 16th Asian Institute in Statistical Genetics and Genomics Workshop.","headline":"Statistical Learning for Multi-omics Data Analysis with R","mainEntityOfPage":{"@type":"WebPage","@id":"https://dlekdud2502.github.io/dlekdud2502/posts/Statistical-Learning-for-Multi&-45;omics-Data-Analysis-with-R/"},"url":"https://dlekdud2502.github.io/dlekdud2502/posts/Statistical-Learning-for-Multi&-45;omics-Data-Analysis-with-R/"}</script><title>Statistical Learning for Multi&#45;omics Data Analysis with R | Dayoung Lee</title><link rel="apple-touch-icon" sizes="180x180" href="/assets/img/favicons/apple-icon-72x72.png"><link rel="icon" type="image/png" sizes="32x32" href="/assets/img/favicons/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/assets/img/favicons/favicon-16x16.png"><link rel="manifest" href="/assets/img/favicons/manifest.json"><link rel="shortcut icon" href="/assets/img/favicons/favicon.ico"><meta name="apple-mobile-web-app-title" content="Dayoung Lee"><meta name="application-name" content="Dayoung Lee"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/assets/img/favicons/browserconfig.xml"><meta name="theme-color" content="#ffffff"><link rel="preconnect" href="https://fonts.googleapis.com" ><link rel="dns-prefetch" href="https://fonts.googleapis.com" ><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="dns-prefetch" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://fonts.googleapis.com" ><link rel="dns-prefetch" href="https://fonts.googleapis.com" ><link rel="preconnect" href="https://cdn.jsdelivr.net" ><link rel="dns-prefetch" href="https://cdn.jsdelivr.net" ><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Lato&family=Source+Sans+Pro:wght@400;600;700;900&display=swap"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4/dist/css/bootstrap.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.11.2/css/all.min.css"><link rel="stylesheet" href="/assets/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@1.0.1/dist/bootstrap-toc.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/magnific-popup@1/dist/magnific-popup.min.css"> <script src="https://cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js"></script> <script type="text/javascript"> class ModeToggle { static get MODE_KEY() { return "mode"; } static get MODE_ATTR() { return "data-mode"; } static get DARK_MODE() { return "dark"; } static get LIGHT_MODE() { return "light"; } static get ID() { return "mode-toggle"; } constructor() { if (this.hasMode) { if (this.isDarkMode) { if (!this.isSysDarkPrefer) { this.setDark(); } } else { if (this.isSysDarkPrefer) { this.setLight(); } } } let self = this; /* always follow the system prefers */ this.sysDarkPrefers.addEventListener("change", () => { if (self.hasMode) { if (self.isDarkMode) { if (!self.isSysDarkPrefer) { self.setDark(); } } else { if (self.isSysDarkPrefer) { self.setLight(); } } self.clearMode(); } self.notify(); }); } /* constructor() */ get sysDarkPrefers() { return window.matchMedia("(prefers-color-scheme: dark)"); } get isSysDarkPrefer() { return this.sysDarkPrefers.matches; } get isDarkMode() { return this.mode === ModeToggle.DARK_MODE; } get isLightMode() { return this.mode === ModeToggle.LIGHT_MODE; } get hasMode() { return this.mode != null; } get mode() { return sessionStorage.getItem(ModeToggle.MODE_KEY); } /* get the current mode on screen */ get modeStatus() { if (this.isDarkMode || (!this.hasMode && this.isSysDarkPrefer)) { return ModeToggle.DARK_MODE; } else { return ModeToggle.LIGHT_MODE; } } setDark() { $('html').attr(ModeToggle.MODE_ATTR, ModeToggle.DARK_MODE); sessionStorage.setItem(ModeToggle.MODE_KEY, ModeToggle.DARK_MODE); } setLight() { $('html').attr(ModeToggle.MODE_ATTR, ModeToggle.LIGHT_MODE); sessionStorage.setItem(ModeToggle.MODE_KEY, ModeToggle.LIGHT_MODE); } clearMode() { $('html').removeAttr(ModeToggle.MODE_ATTR); sessionStorage.removeItem(ModeToggle.MODE_KEY); } /* Notify another plugins that the theme mode has changed */ notify() { window.postMessage({ direction: ModeToggle.ID, message: this.modeStatus }, "*"); } } /* ModeToggle */ const toggle = new ModeToggle(); function flipMode() { if (toggle.hasMode) { if (toggle.isSysDarkPrefer) { if (toggle.isLightMode) { toggle.clearMode(); } else { toggle.setLight(); } } else { if (toggle.isDarkMode) { toggle.clearMode(); } else { toggle.setDark(); } } } else { if (toggle.isSysDarkPrefer) { toggle.setLight(); } else { toggle.setDark(); } } toggle.notify(); } /* flipMode() */ </script><body data-spy="scroll" data-target="#toc" data-topbar-visible="true"><div id="sidebar" class="d-flex flex-column align-items-end"><div class="profile-wrapper text-center"><div id="avatar"> <a href="/" class="mx-auto"> <img src="/assets/profile.jpg" alt="avatar" onerror="this.style.display='none'"> </a></div><div class="site-title mt-3"> <a href="/">Dayoung Lee</a></div><div class="site-subtitle font-italic">Bioinformatics</div></div><ul class="w-100"><li class="nav-item"> <a href="/" class="nav-link"> <i class="fa-fw fas fa-home ml-xl-3 mr-xl-3 unloaded"></i> <span>HOME</span> </a><li class="nav-item"> <a href="/categories/" class="nav-link"> <i class="fa-fw fas fa-stream ml-xl-3 mr-xl-3 unloaded"></i> <span>CATEGORIES</span> </a><li class="nav-item"> <a href="/tags/" class="nav-link"> <i class="fa-fw fas fa-tag ml-xl-3 mr-xl-3 unloaded"></i> <span>TAGS</span> </a><li class="nav-item"> <a href="/archives/" class="nav-link"> <i class="fa-fw fas fa-archive ml-xl-3 mr-xl-3 unloaded"></i> <span>ARCHIVES</span> </a><li class="nav-item"> <a href="/about/" class="nav-link"> <i class="fa-fw fas fa-info-circle ml-xl-3 mr-xl-3 unloaded"></i> <span>ABOUT</span> </a></ul><div class="sidebar-bottom mt-auto d-flex flex-wrap justify-content-center align-items-center"> <button class="mode-toggle btn" aria-label="Switch Mode"> <i class="fas fa-adjust"></i> </button> <span class="icon-border"></span> <a href="https://github.com/dlekdud2502" aria-label="github" target="_blank" rel="noopener"> <i class="fab fa-github"></i> </a> <a href=" javascript:location.href = 'mailto:' + ['ldy990512','gmail.com'].join('@')" aria-label="email" > <i class="fas fa-envelope"></i> </a></div></div><div id="topbar-wrapper"><div id="topbar" class="container d-flex align-items-center justify-content-between h-100 pl-3 pr-3 pl-md-4 pr-md-4"> <span id="breadcrumb"> <span> <a href="/"> Home </a> </span> <span>Statistical Learning for Multi&#45;omics Data Analysis with R</span> </span> <i id="sidebar-trigger" class="fas fa-bars fa-fw"></i><div id="topbar-title"> Post</div><i id="search-trigger" class="fas fa-search fa-fw"></i> <span id="search-wrapper" class="align-items-center"> <i class="fas fa-search fa-fw"></i> <input class="form-control" id="search-input" type="search" aria-label="search" autocomplete="off" placeholder="Search..."> </span> <span id="search-cancel" >Cancel</span></div></div><div id="main-wrapper" class="d-flex justify-content-center"><div id="main" class="container pl-xl-4 pr-xl-4"><div class="row"><div id="core-wrapper" class="col-12 col-lg-11 col-xl-9 pr-xl-4"><div class="post pl-1 pr-1 pl-md-2 pr-md-2"><h1 data-toc-skip>Statistical Learning for Multi&#45;omics Data Analysis with R</h1><div class="post-meta text-muted"> <span> Posted <em class="" data-ts="1658811600" data-df="ll" data-toggle="tooltip" data-placement="bottom"> Jul 26, 2022 </em> </span> <span> Updated <em class="" data-ts="1658817240" data-df="ll" data-toggle="tooltip" data-placement="bottom"> Jul 26, 2022 </em> </span><div class="d-flex justify-content-between"> <span> By <em> <a href="https://github.com/dlekdud2502">Dayoung Lee</a> </em> </span><div> <span class="readtime" data-toggle="tooltip" data-placement="bottom" title="593 words"> <em>3 min</em> read</span></div></div></div><div class="post-content"><p>This post is written after taking lecture “Multi-omics Data Analysis” of 16th Asian Institute in Statistical Genetics and Genomics Workshop.</p><h2 id="day-1-21jul2022"><span class="mr-2">DAY 1 (21.Jul.2022)</span></h2><h3 id="1-dimensionality-reduction"><span class="mr-2">1. Dimensionality Reduction</span></h3><p>Convert high-dimensional data to low-dimentional data.</p><p>1. Removal of uninformative features based on experience.<br /> 2. Removal of uninformative features based on computational method. ex) PCA</p><h4 id="1-1-principal-component-analysis-pca"><span class="mr-2">1-1. Principal Component Analysis (PCA)</span></h4><p>Define y_k that is orthogonal with y_(k-1). Select n of y_k that have largest variance.</p><div class="language-R highlighter-rouge"><div class="code-header"> <span data-label-text="R"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
</pre><td class="rouge-code"><pre><span class="c1"># Example R code of PCA</span><span class="w">
</span><span class="n">X</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">rnorm</span><span class="p">(</span><span class="m">32000</span><span class="p">),</span><span class="w"> </span><span class="m">1000</span><span class="p">,</span><span class="w"> </span><span class="m">32</span><span class="p">)</span><span class="w">
</span><span class="n">pca_X</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">prcomp</span><span class="p">(</span><span class="n">X</span><span class="p">)</span><span class="w">
</span></pre></table></code></div></div><p><img data-src="/assets/posts/220726_fig01.jpg" alt="alt Figure 01" data-proofer-ignore> <em>Figure 01. Example of PCA for TCGA Expression Data</em></p><h4 id="1-2-independent-component-analysis-ica"><span class="mr-2">1-2. Independent Component Analysis (ICA)</span></h4><p>Basic idea is same with PCA, but do not need to be orthgonal.</p><div class="language-R highlighter-rouge"><div class="code-header"> <span data-label-text="R"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
4
5
6
7
8
9
</pre><td class="rouge-code"><pre><span class="c1"># Example R code of ICA</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">fastICA</span><span class="p">)</span><span class="w">
</span><span class="n">s</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">runif</span><span class="p">(</span><span class="m">10000</span><span class="p">),</span><span class="m">5000</span><span class="p">,</span><span class="m">2</span><span class="p">)</span><span class="w">
</span><span class="n">A</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">-1</span><span class="p">,</span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">byrow</span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span><span class="n">X</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">s</span><span class="o">%*%</span><span class="n">A</span><span class="w">

</span><span class="n">a</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fastICA</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">alg.type</span><span class="o">=</span><span class="s2">"parallel"</span><span class="p">,</span><span class="w"> </span><span class="n">fun</span><span class="o">=</span><span class="w"> </span><span class="s2">"logcosh"</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w">
             </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"C"</span><span class="p">,</span><span class="w"> </span><span class="n">row.norm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">maxit</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">200</span><span class="p">,</span><span class="w">
             </span><span class="n">tol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.0001</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span></pre></table></code></div></div><p><img data-src="/assets/posts/220726_fig02.jpg" alt="alt Figure 02" data-proofer-ignore> <em>Figure 02. Difference between PCA and ICA</em></p><h4 id="1-3-non-negative-matrix-factorization-nmf"><span class="mr-2">1-3. Non-negative Matrix Factorization (NMF)</span></h4><p>The matrix V is represented by the two smaller matrices W and H, which, when multiplied, reconstruct V.</p><p><img data-src="/assets/posts/220726_fig03.jpg" alt="alt Figure 03" data-proofer-ignore> <em>Figure 03. NMF</em></p><div class="language-R highlighter-rouge"><div class="code-header"> <span data-label-text="R"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
4
5
6
7
8
9
10
11
</pre><td class="rouge-code"><pre><span class="c1"># Example R code of NMF</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"NMF"</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">NMF</span><span class="p">)</span><span class="w">

</span><span class="n">max</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">runif</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">min</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">max</span><span class="o">=</span><span class="m">100</span><span class="p">),</span><span class="w"> </span><span class="n">nrow</span><span class="o">=</span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">10</span><span class="p">)</span><span class="w">
</span><span class="n">res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">nmf</span><span class="p">(</span><span class="n">mat</span><span class="p">,</span><span class="w"> </span><span class="n">rank</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="w">
</span><span class="n">w</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">basis</span><span class="p">(</span><span class="n">res</span><span class="p">)</span><span class="w">
</span><span class="n">h</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">coef</span><span class="w">

</span><span class="n">w</span><span class="w"> </span><span class="o">%*%</span><span class="w"> </span><span class="n">h</span><span class="w"> </span><span class="c1"># ~mat</span><span class="w">
</span><span class="nf">dim</span><span class="p">(</span><span class="n">w</span><span class="p">)</span><span class="w"> </span><span class="c1"># 30 2</span><span class="w">
</span></pre></table></code></div></div><h4 id="1-4-t-distributed-stochastic-neighbor-embedding-t-sne"><span class="mr-2">1-4. t-distributed Stochastic Neighbor Embedding (t-SNE)</span></h4><p>Nonlinear dimensionality reduction technique. The distribution of data is checked by distance based on t-distribution.</p><p><img data-src="/assets/posts/220726_fig04.jpg" alt="alt Figure 04" data-proofer-ignore> <em>Figure 04. t-SNE</em></p><div class="language-R highlighter-rouge"><div class="code-header"> <span data-label-text="R"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
4
5
6
7
8
9
10
</pre><td class="rouge-code"><pre><span class="c1"># Example R code of t-SNE</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"Rtsne"</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">Rtsne</span><span class="p">)</span><span class="w">
</span><span class="n">data</span><span class="p">(</span><span class="n">iris</span><span class="p">)</span><span class="w">

</span><span class="n">iris_unique</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unique</span><span class="p">(</span><span class="n">iris</span><span class="p">)</span><span class="w">
</span><span class="n">iris_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">iris_unique</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">])</span><span class="w">
</span><span class="n">tsne_out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Rtsne</span><span class="p">(</span><span class="n">iris_matrix</span><span class="p">)</span><span class="w"> 

</span><span class="n">plot</span><span class="p">(</span><span class="n">tsne_out</span><span class="o">$</span><span class="n">Y</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="n">iris_unique</span><span class="o">$</span><span class="n">Species</span><span class="p">)</span><span class="w">
</span></pre></table></code></div></div><p><img data-src="/assets/posts/220726_fig05.jpg" alt="alt Figure 05" data-proofer-ignore> <em>Figure 05. tsne_out</em></p><h3 id="2-clustering"><span class="mr-2">2. Clustering</span></h3><ul><li>No labels<li>Group points into clusters based on how “near” they are to one another<li>Identify structure in data<li>Unsupervised learning</ul><h4 id="2-1-k-means-clustering"><span class="mr-2">2-1. k-means clustering</span></h4><p>1. Ask user how many clusters they’d like. (e.g. k=5)<br /> 2. Randomly guess k cluster center locations<br /> 3. Each datapoint finds out which center it’s closest to.<br /> 4. Each center finds the centroid of the points it owns and jumps there.<br /> 5. Repeat until terminated.</p><p><img data-src="/assets/posts/220726_fig06.jpg" alt="alt Figure 06" data-proofer-ignore> <em>Figure 06. k-means Clustering Algorithm</em></p><div class="language-R highlighter-rouge"><div class="code-header"> <span data-label-text="R"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
4
5
6
7
</pre><td class="rouge-code"><pre><span class="c1"># Example R code of k-means Clustering</span><span class="w">
</span><span class="n">iris2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">iris</span><span class="w">
</span><span class="n">iris2</span><span class="o">$</span><span class="n">Species</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span><span class="w">
</span><span class="n">kmeans.result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">kmeans</span><span class="p">(</span><span class="n">iris2</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w">
</span><span class="n">kmeans.result</span><span class="w"> </span><span class="c1"># Figure 07</span><span class="w">

</span><span class="n">plot</span><span class="p">(</span><span class="n">iris2</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Sepal.Length"</span><span class="p">,</span><span class="s2">"Sepal.Width"</span><span class="p">)],</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="n">kmeans.result</span><span class="o">$</span><span class="n">cluster</span><span class="p">)</span><span class="w"> </span><span class="c1"># Figure 08</span><span class="w">
</span></pre></table></code></div></div><p><img data-src="/assets/posts/220726_fig07.jpg" alt="alt Figure 07" data-proofer-ignore> <em>Figure 07. kmeans.result</em></p><p><img data-src="/assets/posts/220726_fig08.jpg" alt="alt Figure 08" data-proofer-ignore> <em>Figure 08. kmeans.result plot</em></p><h4 id="2-2-hierarchical-clustering"><span class="mr-2">2-2. Hierarchical Clustering</span></h4><p>1. Select two samples and measure distance between them.<br /> 2. Make the nearest two samples into a group and regard them as one sample.<br /> 3. Repeat until terminated.</p><ul><li>How to measure distance between groups?</ul><ol><li>Single Linkage<br /> Dissimilarity between two clusters = Minimum dissimilarity between the members of two clusters.<li>Complete Linkage<br /> Dissimilarity between two clusters = Minimum dissimilarity between the members of two clusters.<li>Average Linkage<br /> Dissimilarity between two clusters = Averaged distances of all pairs of objects (one from each cluster).<li>Average Group Linkage<br /> Dissimilarity between two clusters = Distance between two cluster means.</ol><p><img data-src="/assets/posts/220726_fig09.jpg" alt="alt Figure 09" data-proofer-ignore> <em>Figure 09. Four method to measure distance between groups <br />(a) Single Linkage (b) Complete Linkage (c) Average Linkage (d) Average Group Linkage</em></p><div class="language-R highlighter-rouge"><div class="code-header"> <span data-label-text="R"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
4
5
6
7
8
9
</pre><td class="rouge-code"><pre><span class="c1"># Example R code of Hierachical Clustering</span><span class="w">
</span><span class="n">idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">dim</span><span class="p">(</span><span class="n">iris</span><span class="p">)[</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="m">40</span><span class="p">)</span><span class="w">
</span><span class="n">irisSample</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">iris</span><span class="p">[</span><span class="n">idx</span><span class="p">,]</span><span class="w"> 
</span><span class="n">irisSample</span><span class="o">$</span><span class="n">Species</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span><span class="w">

</span><span class="n">hc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">hclust</span><span class="p">(</span><span class="n">dist</span><span class="p">(</span><span class="n">irisSample</span><span class="p">),</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s2">"ave"</span><span class="p">)</span><span class="w">
</span><span class="n">hc</span><span class="w"> </span><span class="c1"># Figure 10</span><span class="w">

</span><span class="n">plot</span><span class="p">(</span><span class="n">hc</span><span class="p">,</span><span class="w"> </span><span class="n">hang</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span><span class="p">,</span><span class="w"> </span><span class="n">labels</span><span class="o">=</span><span class="n">iris</span><span class="o">$</span><span class="n">Species</span><span class="p">[</span><span class="n">idx</span><span class="p">])</span><span class="w"> </span><span class="c1"># Figure 11</span><span class="w">
</span></pre></table></code></div></div><p><img data-src="/assets/posts/220726_fig10.jpg" alt="alt Figure 10" data-proofer-ignore> <em>Figure 10. hc</em> <img data-src="/assets/posts/220726_fig11.jpg" alt="alt Figure 11" data-proofer-ignore> <em>Figure 11. hc plot</em></p><h2 id="day-2-22jul2022"><span class="mr-2">DAY 2 (22.Jul.2022)</span></h2><p>The contents of Day 2 are summarized in the file below.</p><div type="down_btn"> <a href="/assets/posts/220726.pdf"> <img data-src="/assets/posts/220726_pdf_start.jpg" data-proofer-ignore> </a></div></div><div class="post-tail-wrapper text-muted"><div class="post-meta mb-3"> <i class="far fa-folder-open fa-fw mr-1"></i> <a href='/categories/workshop-conference/'>Workshop/Conference</a>, <a href='/categories/16th-asian-institute-in-statistical-genetics-and-genomics-workshop/'>16th Asian Institute in Statistical Genetics and Genomics Workshop</a></div><div class="post-tags"> <i class="fa fa-tags fa-fw mr-1"></i> <a href="/tags/r/" class="post-tag no-text-decoration" >r</a> <a href="/tags/statistical-learning/" class="post-tag no-text-decoration" >statistical learning</a> <a href="/tags/machine-learning/" class="post-tag no-text-decoration" >machine learning</a> <a href="/tags/workshop/" class="post-tag no-text-decoration" >workshop</a> <a href="/tags/multiomics/" class="post-tag no-text-decoration" >multiomics</a></div><div class="post-tail-bottom d-flex justify-content-between align-items-center mt-3 pt-5 pb-2"><div class="license-wrapper"> This post is licensed under <a href="https://creativecommons.org/licenses/by/4.0/"> CC BY 4.0 </a> by the author.</div><div class="share-wrapper"> <span class="share-label text-muted mr-1">Share</span> <span class="share-icons"> <a href="https://twitter.com/intent/tweet?text=Statistical+Learning+for+Multi%26%2345%3Bomics+Data+Analysis+with+R+-+Dayoung+Lee&url=https%3A%2F%2Fdlekdud2502.github.io%2Fdlekdud2502%2Fposts%2FStatistical-Learning-for-Multi%26-45%3Bomics-Data-Analysis-with-R%2F" data-toggle="tooltip" data-placement="top" title="Twitter" target="_blank" rel="noopener" aria-label="Twitter"> <i class="fa-fw fab fa-twitter"></i> </a> <a href="https://www.facebook.com/sharer/sharer.php?title=Statistical+Learning+for+Multi%26%2345%3Bomics+Data+Analysis+with+R+-+Dayoung+Lee&u=https%3A%2F%2Fdlekdud2502.github.io%2Fdlekdud2502%2Fposts%2FStatistical-Learning-for-Multi%26-45%3Bomics-Data-Analysis-with-R%2F" data-toggle="tooltip" data-placement="top" title="Facebook" target="_blank" rel="noopener" aria-label="Facebook"> <i class="fa-fw fab fa-facebook-square"></i> </a> <a href="https://t.me/share/url?url=https%3A%2F%2Fdlekdud2502.github.io%2Fdlekdud2502%2Fposts%2FStatistical-Learning-for-Multi%26-45%3Bomics-Data-Analysis-with-R%2F&text=Statistical+Learning+for+Multi%26%2345%3Bomics+Data+Analysis+with+R+-+Dayoung+Lee" data-toggle="tooltip" data-placement="top" title="Telegram" target="_blank" rel="noopener" aria-label="Telegram"> <i class="fa-fw fab fa-telegram"></i> </a> <i id="copy-link" class="fa-fw fas fa-link small" data-toggle="tooltip" data-placement="top" title="Copy link" data-title-succeed="Link copied successfully!"> </i> </span></div></div></div></div></div><div id="panel-wrapper" class="col-xl-3 pl-2 text-muted"><div class="access"><div id="access-lastmod" class="post"><div class="panel-heading">Recently Updated</div><ul class="post-content pl-0 pb-1 ml-1 mt-2"><li><a href="/posts/Statistical-Learning-for-Multi&-45;omics-Data-Analysis-with-R/">Statistical Learning for Multi&#45;omics Data Analysis with R</a><li><a href="/posts/How-to-Write-a-Post-&-45;-Markdown-Tutorial-2/">How to Write a Post &#45; Markdown Tutorial 2</a><li><a href="/posts/DEG-Analysis-with-limma/">DEG Analysis with limma</a><li><a href="/posts/How-to-Write-a-Post-&-45;-Markdown-Tutorial-1/">How to Write a Post &#45; Markdown Tutorial 1</a></ul></div><div id="access-tags"><div class="panel-heading">Trending Tags</div><div class="d-flex flex-wrap mt-3 mb-1 mr-3"> <a class="post-tag" href="/tags/chirpy/">chirpy</a> <a class="post-tag" href="/tags/github/">github</a> <a class="post-tag" href="/tags/jekyll/">jekyll</a> <a class="post-tag" href="/tags/markdown/">markdown</a> <a class="post-tag" href="/tags/r/">r</a> <a class="post-tag" href="/tags/deg/">deg</a> <a class="post-tag" href="/tags/limma/">limma</a> <a class="post-tag" href="/tags/machine-learning/">machine learning</a> <a class="post-tag" href="/tags/multiomics/">multiomics</a> <a class="post-tag" href="/tags/statistical-learning/">statistical learning</a></div></div></div><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@1.0.1/dist/bootstrap-toc.min.js"></script><div id="toc-wrapper" class="pl-0 pr-4 mb-5"><div class="panel-heading pl-3 pt-2 mb-2">Contents</div><nav id="toc" data-toggle="toc"></nav></div></div></div><div class="row"><div id="tail-wrapper" class="col-12 col-lg-11 col-xl-9 pl-3 pr-3 pr-xl-4"><div id="related-posts" class="mt-5 mb-2 mb-sm-4"><h3 class="pt-2 mt-1 mb-4 ml-1" data-toc-skip>Further Reading</h3><div class="card-deck mb-4"><div class="card"> <a href="/posts/DEG-Analysis-with-limma/"><div class="card-body"> <em class="small" data-ts="1656645720" data-df="ll" > Jul 1, 2022 </em><h3 class="pt-0 mt-1 mb-3" data-toc-skip>DEG Analysis with limma</h3><div class="text-muted small"><p> 1) What is “limma”? It is an R/Bioconductor software package that provides an integrated solution for analysing data from gene expression expeirments. For limma, both RNA-seq and microarray data c...</p></div></div></a></div><div class="card"> <a href="/posts/How-to-Write-a-Post-&-45;-Markdown-Tutorial-2/"><div class="card-body"> <em class="small" data-ts="1657698900" data-df="ll" > Jul 13, 2022 </em><h3 class="pt-0 mt-1 mb-3" data-toc-skip>How to Write a Post &#45; Markdown Tutorial 2</h3><div class="text-muted small"><p> 1) Picture ![alt &quot;Figure explanation&quot;](&quot;File path&quot;) If you want to insert caption, use *caption*.      2) Table   2.1) Structure of Table // Space is neccesary before table | First Header | S...</p></div></div></a></div><div class="card"> <a href="/posts/How-to-Write-a-Post-&-45;-Markdown-Tutorial-1/"><div class="card-body"> <em class="small" data-ts="1656045540" data-df="ll" > Jun 24, 2022 </em><h3 class="pt-0 mt-1 mb-3" data-toc-skip>How to Write a Post &#45; Markdown Tutorial 1</h3><div class="text-muted small"><p> 0) Write Markdown Files for Posts The title of the .md(markdown) file for post should be &quot;YYYY-MM-DD-TITLE.md&quot; and locate the file in the _post/ folder. You should start your file with the contents...</p></div></div></a></div></div></div><div class="post-navigation d-flex justify-content-between"> <a href="/posts/How-to-Write-a-Post-&-45;-Markdown-Tutorial-2/" class="btn btn-outline-primary" prompt="Older"><p>How to Write a Post &#45; Markdown Tutorial 2</p></a><div class="btn btn-outline-primary disabled" prompt="Newer"><p>-</p></div></div></div></div><footer class="row pl-3 pr-3"><div class="col-12 d-flex justify-content-between align-items-center text-muted pl-0 pr-0"><div class="footer-left"><p class="mb-0"> © 2022 <a href="https://github.com/dlekdud2502">Dayoung Lee</a>. <span data-toggle="tooltip" data-placement="top" title="Except where otherwise noted, the blog posts on this site are licensed under the Creative Commons Attribution 4.0 International (CC BY 4.0) License by the author.">Some rights reserved.</span></p></div><div class="footer-right"><p class="mb-0"> Powered by <a href="https://jekyllrb.com" target="_blank" rel="noopener">Jekyll</a> with <a href="https://github.com/cotes2020/jekyll-theme-chirpy" target="_blank" rel="noopener">Chirpy</a> theme.</p></div></div></footer></div><div id="search-result-wrapper" class="d-flex justify-content-center unloaded"><div class="col-12 col-sm-11 post-content"><div id="search-hints"><div id="access-tags"><div class="panel-heading">Trending Tags</div><div class="d-flex flex-wrap mt-3 mb-1 mr-3"> <a class="post-tag" href="/tags/chirpy/">chirpy</a> <a class="post-tag" href="/tags/github/">github</a> <a class="post-tag" href="/tags/jekyll/">jekyll</a> <a class="post-tag" href="/tags/markdown/">markdown</a> <a class="post-tag" href="/tags/r/">r</a> <a class="post-tag" href="/tags/deg/">deg</a> <a class="post-tag" href="/tags/limma/">limma</a> <a class="post-tag" href="/tags/machine-learning/">machine learning</a> <a class="post-tag" href="/tags/multiomics/">multiomics</a> <a class="post-tag" href="/tags/statistical-learning/">statistical learning</a></div></div></div><div id="search-results" class="d-flex flex-wrap justify-content-center text-muted mt-3"></div></div></div></div><div id="mask"></div><a id="back-to-top" href="#" aria-label="back-to-top" class="btn btn-lg btn-box-shadow" role="button"> <i class="fas fa-angle-up"></i> </a><div id="notification" class="toast" role="alert" aria-live="assertive" aria-atomic="true" data-animation="true" data-autohide="false"><div class="toast-header"> <button type="button" class="ml-2 ml-auto close" data-dismiss="toast" aria-label="Close"> <span aria-hidden="true">&times;</span> </button></div><div class="toast-body text-center pt-0"><p class="pl-2 pr-2 mb-3">A new version of content is available.</p><button type="button" class="btn btn-primary" aria-label="Update"> Update </button></div></div><script src="https://cdn.jsdelivr.net/npm/simple-jekyll-search@1.10.0/dest/simple-jekyll-search.min.js"></script> <script> SimpleJekyllSearch({ searchInput: document.getElementById('search-input'), resultsContainer: document.getElementById('search-results'), json: '/assets/js/data/search.json', searchResultTemplate: '<div class="pl-1 pr-1 pl-sm-2 pr-sm-2 pl-lg-4 pr-lg-4 pl-xl-0 pr-xl-0"> <a href="{url}">{title}</a><div class="post-meta d-flex flex-column flex-sm-row text-muted mt-1 mb-1"> {categories} {tags}</div><p>{snippet}</p></div>', noResultsText: '<p class="mt-5">Oops! No results found.</p>', templateMiddleware: function(prop, value, template) { if (prop === 'categories') { if (value === '') { return `${value}`; } else { return `<div class="mr-sm-4"><i class="far fa-folder fa-fw"></i>${value}</div>`; } } if (prop === 'tags') { if (value === '') { return `${value}`; } else { return `<div><i class="fa fa-tag fa-fw"></i>${value}</div>`; } } } }); </script> <script src="https://cdn.jsdelivr.net/combine/npm/magnific-popup@1/dist/jquery.magnific-popup.min.js,npm/lozad/dist/lozad.min.js,npm/clipboard@2/dist/clipboard.min.js"></script> <script src="https://cdn.jsdelivr.net/combine/npm/dayjs@1/dayjs.min.js,npm/dayjs@1/locale/en.min.js,npm/dayjs@1/plugin/relativeTime.min.js,npm/dayjs@1/plugin/localizedFormat.min.js"></script> <script defer src="/assets/js/dist/post.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4/dist/js/bootstrap.bundle.min.js"></script> <script defer src="/app.js"></script>
